packages/wonder-blocks-dropdown/src/components/dropdown-core.js

```diff
@@ -661,83 +776,37 @@ class DropdownCore extends React.Component<Props, State> {
-            document.querySelector("body");
-
-        if (modalHost) {
-            return ReactDOM.createPortal(
-                <Popper
-                    innerRef={(node: ?HTMLElement) => {
-                        if (node) {
-                            this.popperElement = node;
-                        }
-                    }}
-                    referenceElement={this.props.openerElement}
-                    strategy="fixed"
-                    placement={
-                        alignment === "left" ? "bottom-start" : "bottom-end"
-                    }
-                    modifiers={[
-                        {
-                            name: "preventOverflow",
-                            options: {
-                                rootBoundary: "viewport",
-                                // Allows to overlap the popper in case there's
-                                // no more vertical room in the viewport.
-                                altAxis: true,
-                                // Also needed to make sure the Popper will be
-                                // displayed correctly in different contexts
-                                // (e.g inside a Modal)
-                                tether: false,
-                            },
-                        },
-                    ]}
-                >
-                    {({
-                        placement,
-                        ref,
-                        style,
-                        hasPopperEscaped,
-                        isReferenceHidden,
-                    }) => {
-                        // For some reason react-popper includes `pointerEvents: "none"`
-                        // in the `style` it passes to us, but only when running the tests.
-                        const {pointerEvents: _, ...restStyle} = style;
-                        return (
-                            <div
-                                ref={ref}
-                                style={restStyle}
-                                data-placement={placement}
-                            >
-                                {this.renderItems(
-                                    hasPopperEscaped || isReferenceHidden,
-                                )}
-                            </div>
-                        );
-                    }}
-                </Popper>,
-                modalHost,
-            );
-        }
 
-        return null;
+        // Preprocess the items that are used inside the Popper instance. By
+        // doing this, we optimize the list to be processed only one time
+        // instead of every time popper changes.
+        // NOTE: This improves the performance impact of the dropdown by
+        // reducing the execution time up to 2.5X.
+        const listRenderer = this.shouldVirtualizeList()
+            ? this.renderVirtualizedList()
+            : this.renderList();
```

Comment:
I wonder if memoizing some of this computation would result in further perf gains.  No action req'd for this PR.

Reply:
Good point! I made some initial explorations here, but decided to leave it for the next phase (see a [comment above](https://github.com/Khan/wonder-blocks/pull/1385/files#diff-2f8aa6cd2de3d71f2a379edd27259800ac222fa77bbee3584ff143e64402c2f5R37) pointing to that JIRA ticket :)). I'll add this note to the ticket!

One point to clarify, do you see this as memoizing the computation of the logic (`React.useMemo()`) or memoizing rendering the elements (`React.memo()`)? What do you think what model would fit better for this scenario?

Reply:
That's a good question.  I think `React.memo()` avoids re-renders if the props are the same, but doesn't do anything for state changes (I could be wrong).  `React.useMemo()` seems like a better fit, but would require migration to hooks.  Maybe something to revisit in the future as wonder-blocks gets more hook-ified. 🙂